### PR TITLE
Overriding query parameter name

### PR DESCRIPTION
**Pull request by boggard**
Relates to #1183

### CODE REVIEW

File: core/src/main/java/feign/ParamAlias.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+/**
+ * Allows specify query parameter name different from field name, when Object is used like parameters container
+ *
+ * @see feign.querymap.BeanQueryMapEncoder
+ * @see feign.querymap.FieldQueryMapEncoder
+ */
+@Retention(RUNTIME)
+@Target({ElementType.FIELD, ElementType.METHOD})
+public @interface ParamAlias {
```
Comment (velo) [inline core/src/main/java/feign/ParamAlias.java]: Any particular reason no to use existing `@Param`

Comment (kdavisk6) [reply core/src/main/java/feign/ParamAlias.java]: That would be my recommendation as well.  Consider using our existing `@Param` annotation.

**Comment by kdavisk6** [comment]
Please add a description referencing the issue this PR is for.

**Comment by boggard** [comment]
I used `@Param` following the comments above. Do i need to use `Param.Expander` for parameter value, when `@Param` used?

**Pull request merged**